roles/ovirt-cluster-upgrade/tasks/main.yml

```diff
@@ -1,4 +1,10 @@
 ---
+## https://github.com/ansible/ansible/issues/22397
+## Ansible 2.3 generates a WARNING when using {{ }} in defaults variables of role
+## this workarounds it until Ansible resolves the issue:
+- set_fact:
+    stop_pinned_to_host_vms: "{{ stop_pinned_to_host_vms }}"
+
```

Comment:
Does that mean that we should do that hack in all our existing roles?

Reply:
We don't use aliases in any of our current roles.

Reply:
The reason for the above warning is that 'when' automatically implies '{{ ... }}' it's not going to get resolved.

What exactly were you trying to work around that resulted in you doing this, I'm guessing a | default might have been more appropriate...

Reply:
@tremble The problem is introducing an alias for role variable, I didn't find any better solutin then having this in `defaults/main.yml`:

```yaml
new_variable: "{{ depraceted_variable | default(false) }}"
```

Reply:
I may be fixed, by removing the warning as many people was suggesting. This behavior was in 2.2 and was OK. They've introduced this warning in 2.3

Reply:
Ah, I get it now, you're deliberately forcing the evaluation of deprecated_variable.

IMHO the following is much clearer (assuming you're not sharing the variable across roles):
```
set_fact:
  # deprecated_variable has been replaced by new_variable
  new_variable: "{{ new_variable | default(deprecated_variable) | default (false) }}"
```

The way around you've done this means that the canonical way to reference the desired value within the role is by using the old name, which feels really bad and not like a deprecation at all.

Reply:
Yes thanks, this looks clearer, but what I don't like is, that real default isn't specified in `defaults`, but it's computed in `tasks`. If I will move it to defaults, I will still need to run the `set_fact` tasks, which will evaluate the `new_variable`.